bundles/http/Citrix/Salesforce-Demo/script.js

```diff
@@ -0,0 +1,668 @@
+    {
+        Id: '8',
+        Subject: 'Identify the experts in the company',
+        AccountId: '5',
+        SubmittedOn: twoDaysAgo,
+        DueDate: tomorrow,
+        Priority: 'Low',
+        Status: 'In Progress',
+        Description: "Identify the experts in the company"
+    },
+    {
+        Id: '9',
+        Subject: 'Prepare presentation for new clients',
+        AccountId: '2',
+        SubmittedOn: fiveDaysAgo,
+        DueDate: tomorrow,
+        Priority: 'High',
+        Status: 'Not Started',
+        Description: "Prepare presentation for new clients"
+    },
+    {
+        Id: '10',
+        Subject: 'Send market research to VP',
+        AccountId: '2',
+        SubmittedOn: fiveDaysAgo,
+        DueDate: tomorrow,
+        Priority: 'Medium',
+        Status: 'In Progress',
+        Description: "Send market research to VP"
+    },
+    {
+        Id: '11',
+        Subject: 'Call sales manager and discuss our proposal',
+        AccountId: '2',
+        SubmittedOn: fiveDaysAgo,
+        DueDate: tomorrow,
+        Priority: 'Low',
+        Status: 'Not Started',
+        Description: "Call sales manager and discuss our proposal"
+    }
+]
+
+const priorities = [
+    {
+        Name: 'High',
+        Url: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/ticket_warning.7b6a0a206200e14e1fbf151efd88ab2f.svg'
+    },
+    {
+        Name: 'Medium',
+        Url: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/ticket_move.fb5362498fae9fabdd971b62f0279a69.svg'
+    },
+    {
+        Name: 'Low',
+        Url: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/ticket_comment.624f1825e3e6aede8d689083147dc5e1.svg'
+    }
+]
+
+async function syncAccounts({ dataStore }) {
+    console.log("Synching accounts")
+
+    dataStore.save('accounts', accounts)
+    dataStore.save('opportunities', opportunities)
+    dataStore.save('tasks', tasks)
+    dataStore.save('priorities', priorities)
+
+    console.log("Synchronization finished...")
+}
+
+function generateRandomNumber(floor, ceiling) {
+    return Math.floor(Math.random() * (ceiling - floor)) + floor
+}
+
+async function incrementSyncOpportunity({ dataStore }) {
+    console.log("Running incremental sync...")
+
+    let task = JSON.parse(JSON.stringify(tasks[generateRandomNumber(0, tasks.length)]))
+    task.Id = uuid.v4()
+    task.AccountId = generateRandomNumber(1, accounts.length).toString()
+    task.SubmittedOn = today
+    task.DueDate = todayPlusOneHour
+    task.Subject = "Another " + task.Subject
+    task.Probability = generateRandomNumber(1, 100)
+
+    dataStore.save('tasks', task)
+    console.log("Incremental sync complete")
+}
+
+async function updateTicket({ dataStore, actionParameters }) {
+    console.log(`Updating task with id ${actionParameters.Id}...`)
+
+    const { Id, Subject, AccountId, SubmittedOn, DueDate, Priority, Status, Description } = actionParameters
```

Comment:
could be replaced by [_.pick](https://lodash.com/docs/4.17.15#pick) but then we lose code completion or static analysis checks. It depends on context and developer's mood.